org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/ProjectsManager.java

```diff
@@ -532,13 +534,21 @@ public IStatus runInWorkspace(IProgressMonitor monitor) {
 				try {
 					long start = System.currentTimeMillis();
 					project.refreshLocal(IResource.DEPTH_INFINITE, progress.split(5));
+					URI[] impactedProjectsURIs = null;
 					Optional<IBuildSupport> buildSupport = getBuildSupport(project);
 					if (buildSupport.isPresent()) {
 						buildSupport.get().update(project, force, progress.split(95));
 						registerWatcherJob.schedule();
+						impactedProjectsURIs = buildSupport.get().getAllContainingProjects(project, monitor);
 					}
 					long elapsed = System.currentTimeMillis() - start;
 					JavaLanguageServerPlugin.logInfo("Updated " + projectName + " in " + elapsed + " ms");
+					if (client != null && ArrayUtils.isNotEmpty(impactedProjectsURIs)) {
+						for (URI uri : impactedProjectsURIs) {
+							ActionableNotification notification = new ActionableNotification().withSeverity(MessageType.Log).withMessage(CLASSPATH_UPDATED_NOTIFICATION).withData(uri.toString());
+							client.sendActionableNotification(notification);
+						}
+					}
```

Comment:
The problem here is, you're missing classpath updates when the "Update project configuration" command is not called. For instance, if the dependencies in a maven pom change, classpath will be updated even without "Update project configuration".
The proper way to fix it is to register a global classpath listener and emit classpath change events from there. See https://github.com/redhat-developer/quarkus-ls/blob/b410a0a3dbefd667f04d4e60d0d153fbc9fd6950/microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/MicroProfilePropertiesListenerManager.java#L62-L188